projects/br_scs/src/Mirza/BusinessRegistry/Auth.hs

```diff
@@ -85,3 +87,36 @@ getUserByOAuthSubQuery oauthSub = do
   case r of
     [user] -> pure $ Just user
     _      -> pure Nothing
+
+
+--------------------------------------------------------------------------------
+-- Authorisation
+--------------------------------------------------------------------------------
+
+userOrganisationAutherisation :: ( Member context '[]
+                                 , Member err     '[AsBRError])
+                              => AuthUser
+                              -> GS1CompanyPrefix
+                              -> DB context err OrganisationMapping
+userOrganisationAutherisation (AuthUser (BT.UserId uId)) gs1CompantPrefix = do
+  maybeMapping <- pg $ runSelectReturningOne $ select $ do
+    mapping <- all_ (_organisationMapping businessRegistryDB)
+    guard_ (organisation_mapping_user_id mapping ==. val_ (Schema.UserId uId))
+    guard_ (organisation_mapping_gs1_company_prefix mapping ==. val_ (BizId gs1CompantPrefix))
+    pure $ mapping
+  case maybeMapping of
+    Nothing -> throwing _OperationNotPermittedBRE (gs1CompantPrefix, BT.UserId uId)
+    Just mapping -> pure mapping
+
+
+-- This doesn't really belong anywhere atm, so for now it can go here, but can
+-- be moved somewhere better when a suitable location is found.
+checkUserExistsQuery :: (AsBRError err)
```

Comment:
is `Mirza.BusinessRegistry.Handlers.Users` a better location for this function?
It probably is more of a util than a handler though, so I see your point.

Reply:
Yeah I considered there to, but as it doesn't directly relate to a handler I thought it belonged more tightly with the auth and can always move it later if a better option presents itself. As a general rule I don't like just general Util groupings unless it makes explicit sense (i.e. the grouping makes sense, rather then just a file of leftovers that don't belong anywhere).